### ISSUE TITLE
NewModuleTest: merge `check_jacobian` with `check_gradgrad`

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#49422]
## 🚀 Feature

Currently, in our autogenerated torch.nn test suite, NewModuleTest does the following:
- In `check_jacobian`, we compute the analytical and numerical jacobian and compare them.
- For `check_gradgrad`, we run gradcheck and gradgradcheck. gradcheck computes the analytical and numeric jacobian and compares them.

This seems redundant, and in fact, we have a TODO to remove the redundancy between all of these:

https://github.com/pytorch/pytorch/blob/5912316cf7cd8892c76d9610803f11fbeb880ea6/torch/testing/_internal/common_nn.py#L5043

## Alternatives

Deprecate NewModuleTest, design something saner. 

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @mruberry

**Comment by zou3519** [comment]
I looked into this a little bit. There's an edge case around how nn.Embedding and nn.EmbeddingBag is tested which means we can't straight up delete `check_jacobian` in favor of gradcheck yet. In particular, nn.Embedding(sparse=True) takes in a dense input and weight but gives sparse gradients: gradcheck doesn't know how to handle this case and assumes that the grads must have the same density as the inputs, but `check_jacobian` handles that specifically.

Maybe we should add yet another flag to gradcheck to handle this case. Something less verbose than `dense_inputs_but_sparse_grad`

**Comment by albanD** [comment]
It sounds niche enough that we can use this custom jacobian only for these Modules and use the regular one for all the others. And never do both :D
